src/linker/Linker.Steps/UnreachableBlocksOptimizer.cs

```diff
@@ -1403,6 +1412,8 @@ public bool Analyze ()
 					case Code.Nop:
 						continue;
 					case Code.Pop:
+						if (stack_instr == null)
+							Debug.Fail ("Invalid IL?");
 						stack_instr.Pop ();
```

Comment:
Should we catch the `InvalidOperationException` here and wrap it with a linker exception instead (and add the "Invalid IL message")?

Reply:
I think we already do in https://github.com/dotnet/linker/blob/640878adf1e27cb79bea0a894033f85a84db208d/src/linker/Linker.Steps/MarkStep.cs#L515-L517. Is that what you had in mind?

Reply:
If `stack_instr == null` then `stack_instr.Pop()` will throw `InvalidOperationException`. I was thinking about throwing that with the "invalid IL?" message so that it's displayed to user. I don't mind keeping it this way since I wouldn't expect this to be hit much and either way the stack trace would give a good clue that there's something weird with the IL.